Makefile.core.mk

```diff
@@ -159,178 +135,167 @@ census_packages = \
+	api/authentication/v1beta1/generated.proto \
+	api/authorization/v1/generated.proto \
+	api/authorization/v1beta1/generated.proto \
+	api/autoscaling/v1/generated.proto \
+	api/autoscaling/v2beta1/generated.proto \
+	api/autoscaling/v2beta2/generated.proto \
+	api/batch/v1/generated.proto \
+	api/batch/v1beta1/generated.proto \
+	api/batch/v2alpha1/generated.proto \
+	api/certificates/v1beta1/generated.proto \
+	api/coordination/v1/generated.proto \
+	api/coordination/v1beta1/generated.proto \
+	api/core/v1/generated.proto \
+	api/events/v1beta1/generated.proto \
+	api/extensions/v1beta1/generated.proto \
+	api/imagepolicy/v1alpha1/generated.proto \
+	api/networking/v1/generated.proto \
+	api/networking/v1beta1/generated.proto \
+	api/node/v1alpha1/generated.proto \
+	api/node/v1beta1/generated.proto \
+	api/policy/v1beta1/generated.proto \
+	api/rbac/v1/generated.proto \
+	api/rbac/v1alpha1/generated.proto \
+	api/rbac/v1beta1/generated.proto \
+	api/scheduling/v1/generated.proto \
+	api/scheduling/v1alpha1/generated.proto \
+	api/scheduling/v1beta1/generated.proto \
+	api/settings/v1alpha1/generated.proto \
+	api/storage/v1/generated.proto \
+	api/storage/v1alpha1/generated.proto \
+	api/storage/v1beta1/generated.proto \
+	apimachinery/pkg/runtime/generated.proto \
+	apimachinery/pkg/api/resource/generated.proto \
+	apimachinery/pkg/runtime/schema/generated.proto \
+	apimachinery/pkg/util/intstr/generated.proto \
+	apimachinery/pkg/apis/meta/v1/generated.proto \
+	apimachinery/pkg/apis/meta/v1beta1/generated.proto
+
+k8s_packages = \
+	api/admission/v1 \
+	api/admission/v1beta1 \
+	api/admissionregistration/v1 \
+	api/admissionregistration/v1beta1 \
+	api/apps/v1 \
+	api/apps/v1beta1 \
+	api/apps/v1beta2 \
+	api/auditregistration/v1alpha1 \
+	api/authentication/v1 \
+	api/authentication/v1beta1 \
+	api/authorization/v1 \
+	api/authorization/v1beta1 \
+	api/autoscaling/v1 \
+	api/autoscaling/v2beta1 \
+	api/autoscaling/v2beta2 \
+	api/batch/v1 \
+	api/batch/v1beta1 \
+	api/batch/v2alpha1 \
+	api/certificates/v1beta1 \
+	api/coordination/v1 \
+	api/coordination/v1beta1 \
+	api/core/v1 \
+	api/events/v1beta1 \
+	api/extensions/v1beta1 \
+	api/imagepolicy/v1alpha1 \
+	api/networking/v1 \
+	api/networking/v1beta1 \
+	api/node/v1alpha1 \
+	api/node/v1beta1 \
+	api/policy/v1beta1 \
+	api/rbac/v1 \
+	api/rbac/v1alpha1 \
+	api/rbac/v1beta1 \
+	api/scheduling/v1 \
+	api/scheduling/v1alpha1 \
+	api/scheduling/v1beta1 \
+	api/settings/v1alpha1 \
+	api/storage/v1 \
+	api/storage/v1alpha1 \
+	api/storage/v1beta1 \
+	apimachinery/pkg/runtime \
+	apimachinery/pkg/api/resource \
+	apimachinery/pkg/runtime/schema \
+	apimachinery/pkg/util/intstr \
+	apimachinery/pkg/apis/meta/v1 \
+	apimachinery/pkg/apis/meta/v1beta1
 
 all: build
 
-$(googleprotobuf_protos): %:
-	@mkdir -p googleprotobuf/google/protobuf
-	@curl -sS $(GOOGLEPROTOBUF_URL)/$@ -o googleprotobuf/$@
-
-$(googleprotobuf_packages): %: $(googleprotobuf_protos)
+TMPDIR=${mkdir -d}
```

Comment:
@geeknoid I may have mentioned `mkdir -d`.  what I had meant was:
```
TMPDIR="$(mktemp -d /tmp/gogobuild-XXXXXXXX)"
```
This will create a secure directory in /tmp with gogobuild- and 8 random charracter suffi.

Reply:
Why is that necessary? What makes it "secure"? And doesn't mktemp already create uniquely named directories?

Reply:
no.  without the `-d` flag, no directory is created.  I suppose the reet doesn't matter.  The directory creation is secure in execution.  A special system call is used to make the file or directory which does not allow someone to symlink attack the directory creation.  I know it seems silly to address security ini the non-runtime portion of the code, however, when code goes through a third-party audit such as those required by the CNCF, this would have been caught, causing rework and a re-release to meet the security audit.

```
Stevens-MacBook-Pro:~ sdake$ mktemp
/var/folders/v6/2nsgyt1s7fxbhnm1xjq9xkj00000gn/T/tmp.8CK29JWK
Stevens-MacBook-Pro:~ sdake$ ls -l /var/folders/v6/2nsgyt1s7fxbhnm1xjq9xkj00000gn/T/tmp.8CK29JWK 
-rw-------  1 sdake  staff  0 Sep  4 05:51 /var/folders/v6/2nsgyt1s7fxbhnm1xjq9xkj00000gn/T/tmp.8CK29JWK
```